- MDEV-37001 Enable building RocksDB on non-Linux aarch64 OSes
  fadfd9ea282078a00bd39dc3b0fa30e6fe72e19e 
- Rocksdb: Add missing getauxval() test for Linux
  6d684b64d57c94899126ceb2bedf1725e016dfaa

Index: storage/rocksdb/CMakeLists.txt
--- storage/rocksdb/CMakeLists.txt.orig
+++ storage/rocksdb/CMakeLists.txt
@@ -33,6 +33,16 @@ IF(HAVE_SCHED_GETCPU)
   ADD_DEFINITIONS(-DHAVE_SCHED_GETCPU=1 -DROCKSDB_SCHED_GETCPU_PRESENT)
 ENDIF()
 
+CHECK_FUNCTION_EXISTS(getauxval  HAVE_AUXV_GETAUXVAL)
+if(HAVE_AUXV_GETAUXVAL)
+  ADD_DEFINITIONS(-DROCKSDB_AUXV_GETAUXVAL_PRESENT)
+endif()
+
+CHECK_FUNCTION_EXISTS(elf_aux_info  HAVE_AUXV_ELF_AUX_INFO)
+if(HAVE_AUXV_ELF_AUX_INFO)
+  ADD_DEFINITIONS(-DROCKSDB_AUXV_ELF_AUX_INFO_PRESENT)
+endif()
+
 IF(WITH_VALGRIND)
   ADD_DEFINITIONS(-DROCKSDB_VALGRIND_RUN=1)
 ENDIF()
@@ -63,15 +73,6 @@ IF (WIN32 AND CMAKE_SIZEOF_VOID_P EQUAL 4)
 ENDIF()
 IF(MSVC_ARM64)
  SKIP_ROCKSDB_PLUGIN("Windows ARM64 not supported")
-ENDIF()
-
-#
-# Also, disable on ARM64 when not Linux
-# Requires submodule update to v6.16.3
-# containing commit https://github.com/facebook/rocksdb/commit/ee4bd4780b321ddb5f92a0f4eb956f2a2ebd60dc
-#
-IF(CMAKE_SYSTEM_PROCESSOR MATCHES "(arm64|aarch64)" AND NOT CMAKE_SYSTEM_NAME STREQUAL "Linux")
-  SKIP_ROCKSDB_PLUGIN("ARM64 disabled on all except Linux")
 ENDIF()
 
 # This plugin needs recent C++ compilers (it is using C++11 features)
